security/pkg/nodeagent/secretfetcher/secretfetcher.go

```diff
@@ -104,10 +115,19 @@ func NewSecretFetcher(ingressGatewayAgent bool, endpoint, CAProviderName string,
 
 	if ingressGatewayAgent {
 		ret.UseCaClient = false
+		ret.ServeFallbackSecret = false
 		cs, err := kube.CreateClientset("", "")
 		if err != nil {
 			fatalf("Could not create k8s clientset: %v", err)
 		}
+		// Check if a fallback secret env variable is set.
+		if secretName := os.Getenv(ingressFallbackSecret); secretName != "" {
```

Comment:
Let's use a default name - if var not set, use something like "fallback" or "fallbacksecret" - if the secret is missing we'll have the old behavior.

Reply:
istio-fallback ?

Reply:
gateway-fallback

Reply:
Done.